pkg/simulator/objects/blockdevices_test.go

```diff
@@ -0,0 +1,1306 @@
+package objects
+
+import (
+	wranglerunstructured "github.com/rancher/wrangler/pkg/unstructured"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"os"
+	"testing"
+)
+
+func TestBlockDevices(t *testing.T) {
+	tmpBlockDevices, err := ioutil.TempFile("/tmp", "block-devices")
+	if err != nil {
+		t.Fatalf("error generate temp file for block devices yaml: %v", err)
+	}
+
+	defer os.Remove(tmpBlockDevices.Name())
+	_, err = tmpBlockDevices.Write([]byte(sampleBlockDevices))
+	if err != nil {
+		t.Fatalf("error writing to temp file %s: %v", tmpBlockDevices.Name(), err)
+	}
+
+	objs, err := GenerateObjects(tmpBlockDevices.Name())
+	if err != nil {
+		t.Fatalf("error reading temp block device file %s %v", tmpBlockDevices.Name(), err)
+	}
+	for _, obj := range objs {
+		unstructObj, err := wranglerunstructured.ToUnstructured(obj)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = cleanupObjects(unstructObj.Object)
+		if err != nil {
+			t.Fatalf("error during cleanup of block device object %v", err)
+		}
+
+		err = objectHousekeeping(unstructObj)
+		if err != nil {
+			t.Fatalf("error during block device housekeeping: %v", err)
+		}
+		// check mountPoint values to ensure they have not been removed when they are null
+		_, ok, err := unstructured.NestedFieldNoCopy(unstructObj.Object, "spec", "fileSystem", "mountPoint")
+		if err != nil {
+			t.Fatalf("error fetching spec.fileSystem.mountPoint on block device: %v ", err)
+		}
+
+		if !ok {
+			t.Fatalf("could not find spec.fileSystem.mountPoint for %s \n%v", unstructObj.GetName(), unstructObj)
+		}
+
+		// check mountPoint in status
+		_, ok, err = unstructured.NestedFieldNoCopy(unstructObj.Object, "status", "deviceStatus", "fileSystem", "mountPoint")
+		if err != nil {
+			t.Fatalf("error fetchign spec.fileSystem.mountPoint on block device: %v", err)
+		}
+
+		if !ok {
+			t.Fatalf("could not find spec.fileSystem.mountPoint for %s", unstructObj.GetName())
+		}
+
+		// check for type in status
+		_, ok, err = unstructured.NestedFieldNoCopy(unstructObj.Object, "status", "deviceStatus", "fileSystem", "type")
+		if err != nil {
+			t.Fatalf("error fetchign spec.fileystem.mountPath on block device: %v", err)
+		}
+
+		if !ok {
+			t.Fatalf("could not find spec.filesystem.mountPath for %s", unstructObj.GetName())
+		}
+	}
+}
+
+const sampleBlockDevices = `
+apiVersion: v1
+items:
+- apiVersion: harvesterhci.io/v1beta1
+  kind: BlockDevice
+  metadata:
+    creationTimestamp: "2022-01-15T20:06:43Z"
+    finalizers:
+    - wrangler.cattle.io/harvester-block-device-handler
+    generation: 2833
+    labels:
+      kubernetes.io/hostname: r620-3
+      ndm.harvesterhci.io/device-type: part
+      ndm.harvesterhci.io/parent-device: b2bf49dab59d30c911d1b51b6f464c18
+    managedFields:
+    - apiVersion: harvesterhci.io/v1beta1
+      fieldsType: FieldsV1
+      fieldsV1:
+        f:spec:
+          f:fileSystem:
+            f:forceFormatted: {}
```

Comment:
Is it possible to remove all `manageFields` to reduce lines of code?

Reply:
removed.